### ISSUE TITLE
[Community permission] User "self-report" when they no longer possess the required tokens

### ISSUE THREAD

**Issue by jrainville** [status-im/status-desktop#10095]
### Description

When a user removes tokens from their wallet, we should re-check the community permissions that the user is in, and if they no longer fulfill those conditions, they should "self-report" to the owner node, ie leave the community.

We could add another api/signal that is specifically for that like `leaveTheCommunityBecauseINoLongerFulfillsThePermissions`, but if `leaveCommunity` does the job correctly, might as well reuse it.

I propose to just do it when we do the actions from the wallet itself and not  do periodic checks on the balances.
Firstly, because we don't want to do calls for no reason and secondly because the admin node already does those calls anyway.
I'm open to change my mind on this point though.

Bonus points if we show a screen or notifications to tell them that they where removed from the community.

It would be even better if **before** doing the wallet action that would remove the fulfilled permission, we had a warning, but that's in Wallet territory cc @alaibe

**Comment by mprakhov** [comment]
> @0x-r4bbit @mprakhov do we already do this? Should I close this issue?

On status-go side, we have a logic to check if the user fit the admin/member permissions:
1. on Permission added/removed/changed
2. validate with some interval

If the user does not fit the requirements, we will be kicked or Rejected (not sure which status he will have)

On UI will be shown the windows with permissions required for the community. This is related community admin/member permissions, about channel permissions not sure

Anyway, better to test :)

**Comment by 0x-r4bbit** [comment]
> I propose to just do it when we do the actions from the wallet itself and not do periodic checks on the balances.
Firstly, because we don't want to do calls for no reason and secondly because the admin node already does those calls anyway. I'm open to change my mind on this point though.

Given that the owner node is already performing a check on a regular basis (when creating/editing/deleting permissions + every hour), **and** the user's node is actually performing the same check on the same events + when the wallet updates, or the collectibles change, and, if the permissions are not longer satisfied, the UI prevents users from entering/view/posting in the UI, I think we should rely on that and not actually send additional messages to the owner node.

The only downside of that is tha there's a slight delay community description, as the member list will only be updated after the owner node has indeed performed the check. But I think that's fine.

**Comment by 0x-r4bbit** [comment]
Maybe @cammellos wants to chime in as well.

TLDR; I don't think user/member nodes need to signal to owner/admins that they aren't satisfying token permissions anymore.

**Comment by cammellos** [comment]
@0x-r4bbit I would agree with you, unless there's a good reason on why is necessary.

It might save some checks on the admin side, but as it stands, not sure it would, since we expect the admin to be more online than users, so their check will happen most of the times earlier. 

I don't have a strong opinion on the matter though, either are ok, but if it's just saving calls to the admin, that would be strictly an optimization, and I think we are not quite at that stage yet, so I'd hold off until we get a better idea of the performance profiles/bottlenecks, I would focus for now on functional requirements.

**Comment by jrainville** [comment]
Alright, I'm gonna close this. If we see it as a requirement later, we can re-open.
